Firestore/core/src/firebase/firestore/remote/remote_objc_bridge.mm

```diff
@@ -272,29 +273,29 @@ bool IsLoggingEnabled() {
   return ConvertToByteBuffer([request data]);
 }
 
-std::vector<FSTMaybeDocument*> DatastoreSerializer::MergeLookupResponses(
+std::vector<MaybeDocument> DatastoreSerializer::MergeLookupResponses(
     const std::vector<grpc::ByteBuffer>& responses, Status* out_status) const {
   // Sort by key.
-  std::map<DocumentKey, FSTMaybeDocument*> results;
+  std::map<DocumentKey, MaybeDocument> results;
 
   for (const auto& response : responses) {
     auto* proto = ToProto<GCFSBatchGetDocumentsResponse>(response, out_status);
     if (!out_status->ok()) {
       return {};
     }
-    FSTMaybeDocument* doc = [serializer_ decodedMaybeDocumentFromBatch:proto];
-    results[doc.key] = doc;
+    MaybeDocument doc = [serializer_ decodedMaybeDocumentFromBatch:proto];
+    results[doc.key()] = doc;
```

Comment:
Looking at this line with fresh eyes, I became worried that there might be undefined behavior due to indeterminate evaluation order of arguments. Rewriting this to remove the operator syntax sugar:
```cpp
results.operator[](doc.key()).operator=(std::move(doc));
```
it is not guaranteed that `doc.key()` is evaluated before `std::move(doc)`. However, I believe this code is actually valid, and that is because `std::move` doesn't actually move anything -- it's just a cast, and the move only happens when `operator=` is evaluated, which is guaranteed to happen after `operator[]` is evaluated.

If you feel like this is too subtle, feel free to undo the change, and sorry for making such a tricky suggestion.

Reply:
So I looked at this when you suggested it and convinced myself it works. The sequence of events this code describes is:

  * doc.key() is evaluated and passed to operator[]
  * operator[] default constructs an instance and returns a mutable reference to it
  * operator= move assigns into the defaulted instance

Intuitively the instance into which the assignment is done cannot exist without having accessed the key already.

If a two-argument `operator[]=` existed I think the standard indeterminate sequencing among function arguments would apply. However, that doesn't exist: there's only `operator[]` and `operator=`. This means that the rules on [cppreference](https://en.cppreference.com/w/cpp/language/eval_order) (particularly rule 2 and 8) support my reading.

I'm not certain that in a language lawyer reading of the text this is correct, but in my just past superficial reading I think this is OK.